delayed-data/app.R

```diff
@@ -0,0 +1,99 @@
+library(teal)
+
+data <- teal_data_module(
+  ui = function(id) {
+    ns <- NS(id)
+    fluidPage(
+      selectInput(
+        ns("datasets"),
+        "Choose datasets",
+        choices = c("iris", "mtcars", "CO2", "airquality"),
+        selected = c("iris", "mtcars"),
+        multiple = TRUE
+      ),
+      actionButton(ns("load"), "Load Datasets")
+    )
+  },
+  server = function(id, ...) {
+    moduleServer(id, function(input, output, session) {
+      eventReactive(input$load, {
+        data <- teal.data::teal_data() |>
+          within(
+            {
+              if ("iris" %in% datasets) {
+                iris <- iris
+              }
+              if ("mtcars" %in% datasets) {
+                mtcars <- mtcars
+              }
+              if ("CO2" %in% datasets) {
+                CO2 <- CO2
+              }
+              if ("airquality" %in% datasets) {
+                airquality <- airquality
+              }
+            },
+            datasets = input$datasets
+          )
+        data
+      })
+    })
+  },
+  once = FALSE
+)
+
+filters <- teal_slices(
+  teal_slice(dataname = "mtcars", varname = "gear", multiple = FALSE),
+  teal_slice(dataname = "iris", varname = "Species"),
+  teal_slice(dataname = "iris", varname = "Sepal.Length"),
+  teal_slice(
+    dataname = "iris", varname = "Species3",
+    fixed = TRUE, anchored = FALSE
+  ),
+  teal_slice(
+    dataname = "iris", varname = "Species4",
+    fixed = FALSE, anchored = TRUE, multiple = TRUE
+  ),
+  teal_slice(
+    dataname = "iris", varname = "Species5",
+    fixed = TRUE, anchored = TRUE
+  ),
+  teal_slice(
+    dataname = "iris", id = "custom_expr", title = "Custom Expression",
+    expr = "Sepal.Width > 2.5 & Petal.Length > 1.5",
+  )
+)
```

Comment:
I think filters in context of ddl are irrelevant. There are no filters in custom-transform app you've proposed in this PR and they are comparably irrelevant in both cases. I'd focus on specific functionality rather than showing combination of them - combination is showcased in apps like efficacy or others.